### PR TITLE
mask based multi-index assignment of column values described

### LINKED ISSUE #34383: DOC: Indexing and selecting data and the use of mask/where
#### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#evaluation-order-matters

#### Documentation problem

The documentation talks about how **not** to use `dfb['c'][dfb['a'].str.startswith('o')] = 42` for assignment of multiple values to a `pandas.Series` or `pandas.DataFrame` column. it does however fail to make a concrete suggestion how to do it given the example. 

#### Suggested fix for documentation

I suggest to mention the `pandas.Series.where` and `pandas.Series.mask` methods here in order to show how the assignment can be done, i.e. by adding an example like:

```python
dfb['c'].mask(dfb['a'].str.startswith('o'), 42)
```
which performs the desired assigment of rows matching `o` to `42`. I am happy to contribute this as a PR myself.

### PR DESCRIPTION
**Pull request by psteinb**
- [x] closes #34383
- [x] passes `black pandas`
not sure what this means, will look it up.
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

- ~~[ ] tests added / passed~~ 
this is an PR on documentation only, unit tests are not needed.
- ~~[ ] whatsnew entry~~
I think I don't need this as this is a PR related to documentation only.

### CODE REVIEW

File: doc/source/user_guide/indexing.rst [hunk 1]
```diff
@@ -1870,25 +1870,35 @@ This is the correct access method:
...
 .. ipython:: python
 
-   dfc = pd.DataFrame({'A': ['aaa', 'bbb', 'ccc'], 'B': [1, 2, 3]})
-   dfc.loc[0, 'A'] = 11
-   dfc
+   dfc = pd.DataFrame({'a': ['one', 'one', 'two',
+                             'three', 'two', 'one', 'six'],
+                       'c': np.arange(7)})
+   dfd = dfc.copy()
+
```
Comment (dsaxton) [inline doc/source/user_guide/indexing.rst:1877]: Some extra whitespace around here that should be removed

Comment (psteinb) [reply doc/source/user_guide/indexing.rst]: fixed in https://github.com/pandas-dev/pandas/pull/34461/commits/1bd5d9daa4add2f10bbb959d7e5a6b42df191532

File: doc/source/user_guide/indexing.rst [hunk 2]
```diff
@@ -1870,25 +1870,34 @@ This is the correct access method:
...
+                             'three', 'two', 'one', 'six'],
+                       'c': np.arange(7)})
+   dfd = dfc.copy()
+   # Setting multiple items using a mask (recommended)
+   mask = dfd['a'].str.startswith('o')
+   dfd.loc[mask, 'c'] = 42
+   dfd
+
+   # Setting a single item (recommended)
+   dfd.loc[2, 'a'] = 11
```
Comment (jreback) [inline doc/source/user_guide/indexing.rst:1883]: can you copy first, like L1876.

File: doc/source/user_guide/indexing.rst [hunk 3]
```diff
@@ -1870,25 +1870,34 @@ This is the correct access method:
...
 .. ipython:: python
 
-   dfc = pd.DataFrame({'A': ['aaa', 'bbb', 'ccc'], 'B': [1, 2, 3]})
-   dfc.loc[0, 'A'] = 11
-   dfc
+   dfc = pd.DataFrame({'a': ['one', 'one', 'two',
+                             'three', 'two', 'one', 'six'],
+                       'c': np.arange(7)})
+   dfd = dfc.copy()
+   # Setting multiple items using a mask (recommended)
```
Comment (jreback) [inline doc/source/user_guide/indexing.rst:1877]: you don't need the word recommended here (instead you can use it around L1869)

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/indexing.rst (hunk 3 above)
Comment (psteinb) [reply doc/source/user_guide/indexing.rst:1877]: I put it there as @TomAugspurger suggested it in https://github.com/pandas-dev/pandas/issues/34383 if I understood correctly.

Comment (TomAugspurger) [reply doc/source/user_guide/indexing.rst:1877]: I think Jeff is saying that "This is the correct access method" implies that this is the recommended way to set things.

Perhaps change "This is the correct access method" to something like "Multiple items can be set with `.loc` and a mask."

Comment (psteinb) [reply doc/source/user_guide/indexing.rst:1877]: fine with me.

File: doc/source/user_guide/indexing.rst (hunk 2 above)
Comment (psteinb) [reply doc/source/user_guide/indexing.rst:1884]: fixed in de9f6bd15

File: doc/source/user_guide/indexing.rst (hunk 3 above)
Comment (psteinb) [reply doc/source/user_guide/indexing.rst]: fixed in 5aa53c693

File: doc/source/user_guide/indexing.rst (hunk 2 above)
Comment (psteinb) [reply doc/source/user_guide/indexing.rst:1884]: please resolve at will

File: doc/source/user_guide/indexing.rst (hunk 3 above)
Comment (psteinb) [reply doc/source/user_guide/indexing.rst]: please resolve at will

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @psteinb

**Pull request merged**